src/executeJar.ts

```diff
@@ -240,6 +244,40 @@ export async function loadContextsFromChangelogFile(
       reject(error as Error);
     }
   });
+
+  /**
+   * Sets the context for the cache.
+   *
+   * @param contexts - the contexts that were loaded from the changelog file
+   * @param changelogFile - the absolute path to the changelog file
+   */
+  async function setContextForCache(contexts: string[], changelogFile: string): Promise<void> {
+    const names = await readLiquibaseConfigurationNames();
+    if (!names) {
+      return;
+    }
+    for (const configurationName of names) {
+      const propertyFile = await getPathOfConfiguration(configurationName);
+
+      if (!propertyFile) {
+        continue;
+      }
+
+      const data = readFullValues(configurationName, propertyFile, {
+        defaultDatabaseForConfiguration: getDefaultDatabaseForConfiguration(),
+        liquibaseDirectoryInProject: getLiquibaseFolder(),
+        customDrivers: getCustomDrivers(),
+      });
+
+      if (data.changelogFile.includes(path.relative(getWorkFolder(), changelogFile))) {
```

Comment:
Is your code and logic correct:

Use case:

1. system database with system/changelog.xml
2. data database with data/changelog.xml
3. use "Liquibase: Update" with system database and  system/changelog.xml, load contexts and select one, and then execute the command: cache looks ok
4. use "Liquibase: Update" with data database and select  system/changelog.xml (the same changelog as before), load contexts - **those do not appear in the cache** - and then select a context 

You will have selected contexts disappear (see teams chat for further details)